src/main/java/liquibase/ext/mongodb/statement/CountCollectionByNameStatement.java

```diff
@@ -24,29 +24,23 @@
 import liquibase.nosql.statement.NoSqlQueryForLongStatement;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import org.bson.Document;
 
-import java.util.stream.StreamSupport;
-
+/**
+ * Queries the database for the number of collections that match the supplied collectionName
+ * i.e returns 1 if the collection is present; else 0
+ */
 @Getter
 @EqualsAndHashCode(callSuper = true)
-public class CountCollectionByNameStatement extends AbstractCollectionStatement
+public class CountCollectionByNameStatement extends ListCollectionNamesStatement
```

Comment:
I'm curious - why not prefer composition over inheritance?

Reply:
Just unified logic under AbstractRunCommandStatement as in the end they are runCommand underneath. It eliminates duplication of code and logic like toJs, run, checkErrors, etc. Yes we could think about adopting composition if you give an example of real value. For now I don't see that as an urgency. We could go via SqlGenerators used by over liquibase SqlStatements however think it brings more complexity rather than value.